vendor/blockchain_contracts/src/ethereum/rfc003/erc20_htlc.rs

```diff
@@ -0,0 +1,218 @@
+        token_contract_address: Address,
+        amount: U256,
+    ) -> Self {
+        Self {
+            refund_timestamp,
+            refund_address,
+            redeem_address,
+            secret_hash,
+            token_contract_address,
+            amount,
+        }
+    }
+
+    /// Constructs the payload for funding an `Erc20` HTLC located at the given
+    /// address.
+    pub fn funding_tx_payload(&self, htlc_contract_address: Address) -> Bytes {
+        let transfer_fn_abi = base16!("A9059CBB");
+        let htlc_contract_address = <[u8; 20]>::from(htlc_contract_address);
+        let amount = <[u8; 32]>::from(self.amount);
+
+        let mut data = [0u8; 4 + 32 + 32];
+        data[..4].copy_from_slice(transfer_fn_abi);
+        data[16..36].copy_from_slice(&htlc_contract_address);
+        data[36..68].copy_from_slice(&amount);
+
+        Bytes::from(data.to_vec())
+    }
+
+    pub fn token_contract_address(&self) -> Address {
+        self.token_contract_address
+    }
+
+    pub fn deployment_gas_limit(&self) -> U256 {
+        let bytes: Bytes = self.compile_to_hex().into();
+        let n_bytes = bytes.0.len();
+        let gas_per_byte = 200;
+
+        U256::from(100_000 + n_bytes * gas_per_byte)
+    }
+
+    pub fn tx_gas_limit() -> U256 {
+        U256::from(100_000)
+    }
+
+    pub fn fund_tx_gas_limit() -> U256 {
+        U256::from(100_000)
+    }
+
+    pub fn compile_template_to_hex() -> String {
+        let code_length = Self::CONTRACT_CODE_TEMPLATE.len() / 2; // In hex, each byte is two chars
+
+        let code_length_as_hex = format!("{:0>4x}", code_length);
+
+        let header_length = Self::DEPLOY_HEADER_TEMPLATE.len() / 2;
+        let header_length_as_hex = format!("{:0>4x}", header_length);
+
+        let deploy_header = Self::DEPLOY_HEADER_TEMPLATE
+            .to_string()
+            .replace(
+                Self::CONTRACT_START_POSITION_PLACEHOLDER,
+                &header_length_as_hex,
+            )
+            .replace(Self::CONTRACT_LENGTH_PLACEHOLDER, &code_length_as_hex);
+
+        deploy_header + Self::CONTRACT_CODE_TEMPLATE
+    }
+
+    fn offset(name: &str, placeholder: &str) -> Offset {
+        let contract = Self::compile_template_to_hex();
+
+        let re_match = Regex::new(placeholder)
+            .expect("Could not create regex")
+            .find(contract.as_str())
+            .expect("Could not find placeholder in hex code");
+        Offset::new(
+            name.into(),
+            re_match.start() / 2,
+            re_match.end() / 2,
+            placeholder.len() / 2,
+        )
+    }
+
+    pub fn all_offsets() -> Vec<Offset> {
+        let refund_timestamp = Self::offset("Refund timestamp", Self::REFUND_TIMESTAMP_PLACEHOLDER);
+        let redeem_address = Self::offset("Redeem Address\t", Self::REDEEM_ADDRESS_PLACEHOLDER);
+        let refund_address = Self::offset("Refund Address\t", Self::REFUND_ADDRESS_PLACEHOLDER);
```

Comment:
❌  Rename this stuff to what it's called in RFC003 e.g. `redeem_address` (and remove tabs).
❌ Output in markdown table format (doesn't matter how nice it looks).